org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java

```diff
@@ -77,6 +77,11 @@ private boolean parentProcessStillRunning() {
 				process.destroy();
 				finished = process.waitFor(POLL_DELAY_SECS, TimeUnit.SECONDS); // wait for the process to stop
 			}
+			if (Platform.OS_WIN32.equals(Platform.getOS()) && finished && process.exitValue() > 1) {
+				// the tasklist command should return 0 (parent process exists) or 1 (parent process doesn't exist)
+				JavaLanguageServerPlugin.logInfo("The tasklist command: '" + command + "' returns " + process.exitValue());
```

Comment:
Last time I tried logging anything in that method, it would "activate" the server, making it wait another 30sec before the next polling

Reply:
You can test using the following patch:

```
diff --git a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java
index 597d24e6..6ab798c7 100644
--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java
@@ -64,7 +64,7 @@ public final class ParentProcessWatcher implements Runnable, Function<MessageCon
                }
                String command;
                if (Platform.OS_WIN32.equals(Platform.getOS())) {
-                       command = "cmd /c \"tasklist /FI \"PID eq " + pid + "\" | findstr " + pid + "\"";
+                       command = "cmd /c \"tasklist2 /FI \"PID eq " + pid + "\" | findstr " + pid + "\"";
                } else {
                        command = "ps -p " + pid;
                }
```